inspirehep/records/api/literature.py

```diff
@@ -143,3 +169,133 @@ def _dump_for_es(self):
         serialized_data = super()._dump_for_es()
         serialized_data["_ui_display"] = json.dumps(self.get_ui_data())
         return serialized_data
+
+
+def import_article(identifier: str) -> dict:
+    """Import a new article from arXiv or Crossref based on the identifier.
+
+    This function attempts to parse  and normalize the identifier as a valid
+    arXiv id or DOI. If the identifier is valid and there is no record in
+    Inspire matching the ID, it queries the arXiv/CrossRef APIs and parses
+    the record to make it inspire compliant.
+
+    Args:
+        identifier(str): the ID of the record to import
+
+    Returns:
+        dict: the serialized article
+
+    Raises:
+        ExistingArticleError: if the record is already in Inspire.
+        ImportArticleError: if no article is found.
+        ImportConnectionError: if the importing request fails.
+        ImportParsingError: if an error occurs while parsing the result.
+        UnknownIdentifierError: if the identifier is neither "arxiv" or "doi".
+    """
+    if is_arxiv(identifier):
+        pid_type = "arxiv"
+        pid_value = normalize_arxiv(identifier)
+
+    elif is_doi(identifier):
+        pid_type = "doi"
+        pid_value = normalize_doi(identifier)
+
+    else:
+        raise UnknownImportIdentifierError(identifier)
+
+    pid = PersistentIdentifier.query.filter_by(
+        pid_type=pid_type, pid_value=pid_value
+    ).one_or_none()
+
+    if pid:
+        raise ExistingArticleError(
+            f"Article {identifier} already in Inspire. UUID: {pid.object_uuid}"
+        )
+    importers = {"arxiv": import_arxiv, "doi": import_doi}
+    importer = importers.get(pid_type, UnknownImportIdentifierError)
+    article = importer(pid_value)
+
+    if not article:
+        raise ImportArticleError(f"No article found for {identifier}")
+    return article
+
+
+def import_arxiv(arxiv_id: str) -> dict:
+    """View for retrieving an article from arXiv.
+
+    This endpoint is designed to be queried by inspirehep during an article
+    submission, to auto-fill the submission form if the user provides an arXiv
+    identifier.
+
+    Args:
+        arxiv_id: the normalized arXiv id, e.g. `0804.2273`.
+
+    Returns:
+        dict: a json object with either the parsed article
+
+    Raises:
+        ImportConnectionError: if the request doesn't succeed.
+        ImportParsingError: if any error occurs during the response parsing.
+
+    """
+    url = ARXIV_URL.replace(PLACEHOLDER, arxiv_id)
+
+    try:
+        resp = requests.get(url=url)
+    except Exception as e:
+        raise ImportConnectionError("Cannot contact arXiv")
+
+    if resp.status_code >= 400:
```

Comment:
Do not create two identical exceptions for different errors (at least message should be different)